csharp/src/Ice/InputStream.cs

```diff
@@ -3721,99 +3634,24 @@ private int readInstance(int index, System.Action<Value> cb)
             Value v = null;
             while (true)
             {
-                bool updateCache = false;
```

Comment:
There was surprisingly lots of code to deal with compact IDs. The new version (from ZeroC.Slice) is much simpler.